### PR TITLE
Issue: Checkout Enhancements

### PR DESCRIPTION
**Pull request by mufudzimasaire**
## Overview

Github Issue: https://github.com/shiftcommerce/shift-front-end-react/issues/865

This PR add the `PaymentMethodSummary` component and updates all the renamed components in PR - https://github.com/shiftcommerce/shift-react-components/pull/29. It also addresses the following issues:

- [x] Add Payment method section with an edit button
- [x] Edit button next to Shipping Address for Stripe
- [x] Place order banner should appear on last step of checkout
- [x] Missing padding between 'Continue Shopping' and 'Place Order' buttons
- [x] Clicking edit on payment breaks the 'same as shipping address option'

* **Note:** 
  - `Stripe checkout place order not working` - ticket has been created https://github.com/shiftcommerce/shift-front-end-react/issues/885
  - `Edit button next to Shipping Address which takes you to Paypal to change the shipping address` conclusion from discussion in standup was to keep current implementation, ie. hide edit button and wait for feedback in the next retro

Dependent on Shift React Components PR https://github.com/shiftcommerce/shift-react-components/pull/29

## QA

1. checkout branch `issue/865_checkout_enhancements`

2. Add items to cart and navigate to the first step of checkout - `payment-method` selection

3. Select `Pay By Credit/Debit Card` option and ensure the payment methods summary component is displayed on `Step 2 - Shipping Address`

Payment Methods Summary for Card

4. Navigate to `Step 1 - Payment Methods` and select `PayPal` option. When you are redirected back, ensure that the payment methods summary component is displayed on `Step 2 - Shipping Address`

Payment Methods Summary for PayPal

5. Also check that there is a gap in between the shopping basket buttons - 

Gap Between Buttons

6. Check the Payment Method Summary component on the Review page for both checkout options

Card

PayPal

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -45,7 +45,7 @@
   },
   "dependencies": {
     "@shiftcommerce/shift-node-api": "^0.4.1",
-    "@shiftcommerce/shift-react-components": "^0.11.0",
+    "@shiftcommerce/shift-react-components": "shiftcommerce/shift-react-components#issue/865_checkout_enhancements",
```
Comment (hubertpompecki) [inline package.json]: This PR needs to wait for shift-react-components to go in first.

File: src/actions/checkout-actions.js [hunk 2]
```diff
@@ -103,7 +103,8 @@ export function authorizePayPalAndCreateOrder (payPalOrderID, paymentMethod) {
     // authorize the PayPal Order
     return dispatch(authorizePayPalOrder(payPalOrderID)).then(() => {
       const order = getState().order
-      if (!order.paymentResponseErrors.error.data) {
+      const paymentError = (order.paymentResponseErrors.error && order.paymentResponseErrors.error.data)
+      if (!paymentError) {
```
Comment (hubertpompecki) [inline src/actions/checkout-actions.js]: What happens if we **do** get an error?

File: src/pages/checkout/shipping-method.js [hunk 3]
```diff
@@ -103,7 +104,8 @@ export class ShippingMethodPage extends Component {
     this.setState({ loading: true })
     // update PayPal order total
     return this.props.dispatch(updatePayPalOrderTotal(this.state.payPalOrderID, this.state.purchaseUnitsReferenceID, this.props.cart)).then(() => {
-      if (!this.props.order.paymentResponseErrors.error.data) {
+      const paymentError = (this.props.order.paymentResponseErrors.error && this.props.order.paymentResponseErrors.error.data)
+      if (!paymentError) {
```
Comment (hubertpompecki) [inline src/pages/checkout/shipping-method.js]: What happens if we **do** get an error?

File: src/actions/checkout-actions.js (hunk 2 above)
Comment (mufudzimasaire) [reply src/actions/checkout-actions.js]: its passed to the Payment summary component and displayed as a flash message -

PayPal authorisation error message

File: src/pages/checkout/shipping-method.js (hunk 3 above)
Comment (mufudzimasaire) [reply src/pages/checkout/shipping-method.js]: Same as above, the error thats set in state is passed to the Shipping Methods component and displayed as a flash message -

PayPal authorisation error message

**Pull request merged**